File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/party/PartyAbdicateThroneMessage.scala





















// Generated on 05/03/2013 10:34:46
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait PartyAbdicateThroneMessage extends AbstractPartyMessage {
  override val messageId = 6080

  protected trait Message extends AbstractPartyMessage.Message with Serializable {
    override val messageId = PartyAbdicateThroneMessage.messageId

    val playerId: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(playerId)
    }

  }

  protected def create(partyId: Int, playerId: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val partyId = buf.readInt()
    val playerId = buf.readInt()

    create(partyId, playerId)
  }

}

object PartyAbdicateThroneMessage extends PartyAbdicateThroneMessage {
  case class MessageImpl(partyId: Int, playerId: Int) extends Message
  type Target = MessageImpl

  override def create(partyId: Int, playerId: Int) = MessageImpl(partyId, playerId)

  def apply(partyId: Int, playerId: Int) = create(partyId, playerId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/PartyAbdicateThroneMessage.java


// Generated on 05/08/2013 19:37:48
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PartyAbdicateThroneMessage extends AbstractPartyMessage {
    public static final int MESSAGE_ID = 6080;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int playerId;
    
    public PartyAbdicateThroneMessage() { }
    
    public PartyAbdicateThroneMessage(int partyId, int playerId) {
        super(partyId);
        this.playerId = playerId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(playerId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        playerId = buf.readInt();
        if (playerId < 0)
            throw new RuntimeException("Forbidden value on playerId = " + playerId + ", it doesn't respect the following condition : playerId < 0");
    }
    
}
